app/workers/user_notifications_worker.rb

```diff
@@ -20,11 +20,32 @@ def self.perform
   # Finds all users that registered and need to be approved and schedules a worker
   # to notify all users that could possibly approve him.
   def self.notify_admins_of_new_users
-    users = User.where(approved: false, needs_approval_notification_sent_at: nil)
+    # Temporary solution so that only one of the two activities that are created
+    # when using Shibboleth or LDAP is actually used to send the notification.
+    specific_creations = RecentActivity.where(trackable_type: 'User', notified: [nil, false])
+                               .where("`key` LIKE '%.user.created'").to_a
+    general_creations = RecentActivity.where(trackable_type: 'User', notified: [nil, false], key: 'user.created').to_a
+
+    general_creations.reject! do |gen|
+      if specific_creations.find_index { |spec| spec.trackable_id == gen.trackable_id }
+        gen.update_attributes(notified: true)
+        true
+      else
+        false
+      end
+    end
+
+    activities = specific_creations + general_creations
     recipients = User.where(superuser: true).pluck(:id)
     unless recipients.empty?
-      users.each do |user|
-        Resque.enqueue(UserNeedsApprovalSenderWorker, user.id, recipients)
+      activities.each do |creation|
+        # If user created is a superuser we don't need to send the notification,
+        # and mark it as sent so it doesn't come back to the worker in future queries.
+        if User.find(creation.trackable_id).superuser
```

Comment:
Why exactly we need to check this? The only case when this might happen today is when the first default admin is created when seeding the database.

I think it would make more sense to check whether the user is already approved or not. I already approved, the notification doesn't need to be sent. This might happen in other circumstances, like when users are automatically approved (e.g. when an admin registered the user).